addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java

```diff
@@ -0,0 +1,252 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.avmfritz.handler;
+
+import static org.openhab.binding.avmfritz.BindingConstants.*;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.avmfritz.BindingConstants;
+import org.openhab.binding.avmfritz.config.AvmFritzConfiguration;
+import org.openhab.binding.avmfritz.internal.ahamodel.DeviceModel;
+import org.openhab.binding.avmfritz.internal.ahamodel.SwitchModel;
+import org.openhab.binding.avmfritz.internal.hardware.FritzahaWebInterface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for a FRITZ!Box device. Handles polling of values from AHA devices.
+ * 
+ * @author Robert Bausdorf
+ * 
+ */
+public class BoxHandler extends BaseBridgeHandler 
+implements IFritzHandler {
+	/**
+	 * Logger
+	 */
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+	/**
+	 * the refresh interval which is used to poll values from the fritzaha.
+	 * server (optional, defaults to 15 s)
+	 */
+	private long refreshInterval = 15;
+	/**
+	 * Interface object for querying the FRITZ!Box web interface
+	 */
+	private FritzahaWebInterface connection;
+	/**
+	 * Holder for last data received from the box.
+	 */
+	private Map<String, DeviceModel> deviceList;
+	/**
+	 * Job which will do the FRITZ!Box polling
+	 */
+	private DeviceListPolling pollingRunnable;
+	/**
+	 * Schedule for polling
+	 */
+	private ScheduledFuture<?> pollingJob;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param bridge
+	 *            Bridge object representing a FRITZ!Box
+	 */
+	public BoxHandler(Bridge bridge) {
+		super(bridge);
+		this.deviceList = new TreeMap<String, DeviceModel>();
+		this.pollingRunnable = new DeviceListPolling(this);
+	}
+
+	/**
+	 * Initializes the bridge.
+	 */
+	@Override
+	public void initialize() {
+		logger.debug("About to initialize bridge "
+				+ BindingConstants.BRIDGE_FRITZBOX);
+		Bridge bridge = this.getThing();
+		String soloIp = thing.getUID().getId().replaceAll("[^a-zA-Z0-9.]", ".");
+		// Property is not propagated from dicovery :-(
```

Comment:
It is working - it is filled as a configuration parameter. Once you have done the other two changes I just proposed, the following line works (of course only after line 97):
```
config.getIpAddress()
```